crates/sui-core/src/authority_aggregator.rs

```diff
@@ -962,12 +1073,87 @@ where
             Err(state) => {
                 self.record_process_transaction_metrics(tx_digest, &state);
                 let state = Self::record_non_quorum_effects_maybe(tx_digest, state);
-                Err(QuorumSignTransactionError {
-                    total_stake: self.committee.total_votes,
-                    good_stake: state.good_stake(),
+                Err(self.handle_process_transaction_error(tx_digest, state))
+            }
+        }
+    }
+
+    // TODO make sure this is epoch-boundary-safe
```

Comment:
What is the potential issue here at epoch boundary?

Reply:
Actually this comment existed in QD before I moved all retry logic over to AA. I added this here as a reminder to myself to add a conflicting tx test across epoch boundaries. The test is added now. I don't think there are any other concerns but I will let @longbowlu comment on the original TODO.

Reply:
I think this comment is not relevant any more